include/SFML/System/Vector2.hpp

```diff
@@ -73,6 +73,15 @@ class Vector2
     ////////////////////////////////////////////////////////////
     template <typename U>
     constexpr explicit Vector2(const Vector2<U>& vector);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Construct the vector from a length and angle
+    ///
+    /// \param length Length of vector
+    /// \param phi    Angle off X axis
+    ///
+    ////////////////////////////////////////////////////////////
+    Vector2(T length, Angle phi);
```

Comment:
Do we want to allow negative values of `length`? (as it's currently handled in tests)

* If yes, we should probably rename the parameter, since it's no longer a length and it won't match the getter `length()`. Maybe `r`.
* Also, in that case we should document that it is allowed, and maybe even that `Vector2(r, phi)` is equivalent to `Vector2(-r, phi + 180°)`.
* Should we mention something about `Vector2(r, phi)` being equivalent to `Vector2(r, phi + n * 360°)`? 

Those things may sound obvious, but I think the important message is that the polar angle constructor is lossy, and it may not be able to retrieve the values later as they were provided in the constructor.

Reply:
Furthermore, maybe we should change the `\brief` part to:
```cpp
/// \brief Construct the vector from polar coordinates
```
while for the `(x, y)` constructor, we could say:
```cpp
/// \brief Construct the vector from cartesian coordinates
```

Reply:
> Do we want to allow negative values of length? (as it's currently handled in tests)

In the Tennis example I modified, imagine if you wanted the ball to go backwards along a certain trajectory. In that case it's really helpful to use a negative radius/length instead of having extra logic to flip the angle. Besides, the math is perfectly well defined for all values of `r` and `phi`.

> If yes, we should probably rename the parameter, since it's no longer a length and it won't match the getter length(). Maybe r.

No problem. I'll adopt that r/phi terminology. I think it looks good and I like how it helps avoid confusion with `length()` and `angle()` which as you pointed out won't necessarily return the same values that were provided to this constructor. 

> Should we mention something about Vector2(r, phi) being equivalent to Vector2(r, phi + n * 360°)?

No harm in a little extra documentation :)